itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTxnNotificationLogging.java

```diff
@@ -0,0 +1,191 @@
+    }
+  }
+
+  @After public void tearDown() throws Exception {
+    TestTxnDbUtil.cleanDb(hiveConf);
+    if (hiveDb != null) {
+      hiveDb.close(true);
+    }
+    Hive.closeCurrent();
+    SessionState.get().close();
+  }
+
+   @Test
+   public void testTxnNotificationLogging() throws Exception {
+    try {
+      List<Long> txnIds = openTxns(numberOfTxns, txnType);
+      assertEquals(txnIds.size(), getNumberOfTxns(txnIds, TxnState.OPEN));
+      assertEquals(expectedNotifications,getNumNotifications(txnIds, MessageBuilder.OPEN_TXN_EVENT));
+      Thread.sleep(1000);
+      runHouseKeeperService();
+      if (TxnType.REPL_CREATED != txnType) {
+        assertEquals(txnIds.size(), getNumberOfTxns(txnIds, TxnState.ABORTED));
+        assertEquals(expectedNotifications,getNumNotifications(txnIds, MessageBuilder.ABORT_TXN_EVENT));
+      }
+    } finally {
+      runTxnHouseKeeperService();
+      Hive.closeCurrent();
+    }
+  }
+
+  private int getNumNotifications(List<Long> txnIds, String eventType) throws IOException {
+    EventUtils.NotificationEventIterator eventIterator = getEventsIterator(0, 100, 100);
+    int numNotifications = 0;
+    MessageDeserializer deserializer = null;
+    while (eventIterator.hasNext()) {
+      NotificationEvent ev = eventIterator.next();
+      if (eventType.equals(ev.getEventType())) {
+        deserializer = ReplUtils.getEventDeserializer(ev);
+        switch (ev.getEventType()) {
+        case MessageBuilder.OPEN_TXN_EVENT:
+          OpenTxnMessage openTxnMessage = deserializer.getOpenTxnMessage(ev.getMessage());
+          if(txnIds.contains(openTxnMessage.getTxnIds().get(0))){
+            numNotifications++;
+          }
+          break;
+        case MessageBuilder.ABORT_TXN_EVENT:
+          AbortTxnMessage abortTxnMessage = deserializer.getAbortTxnMessage(ev.getMessage());
+          if(txnIds.contains(abortTxnMessage.getTxnId())){
+            numNotifications++;
+          }
+        }
+      }
+    }
+    return numNotifications;
+  }
+
+  private List<Long> openTxns(int txnCounter, TxnType txnType) throws TException {
+    List<Long> txnIds = new LinkedList<>();
+    for (; txnCounter > 0; txnCounter--) {
+      if (txnType == TxnType.REPL_CREATED) {
+        Long srcTxn = (long) (11 + txnCounter);
+        List<Long> srcTxns = Arrays.asList(new Long[] { srcTxn });
+        txnIds.addAll(hiveDb.getMSC().replOpenTxn("testPolicy", srcTxns, "hive", txnType));
+      } else {
+        txnIds.add(hiveDb.getMSC().openTxn("hive", txnType));
+      }
+    }
+    return txnIds;
+  }
+
+  private int getNumberOfTxns(List<Long> txnIds, TxnState txnState) throws TException {
+    AtomicInteger numTxns = new AtomicInteger();
+    hiveDb.getMSC().showTxns().getOpen_txns().forEach(txnInfo -> {
+      if (txnInfo.getState() == txnState && txnIds.contains(txnInfo.getId())) {
+        numTxns.incrementAndGet();
+      }
+    });
+    return numTxns.get();
+  }
+
+  private void runHouseKeeperService() {
```

Comment:
can you create a base class and move there common methods?

Reply:
This class is not needed now, removed it